ring/partitions_ring_shuffle_shard_cache.go

```diff
@@ -0,0 +1,96 @@
+package ring
+
+import (
+	"math"
+	"sync"
+	"time"
+)
+
+type partitionRingShuffleShardCache struct {
+	mtx                  sync.RWMutex
+	cacheWithoutLookback map[subringCacheKey]*PartitionRing
+	cacheWithLookback    map[subringCacheKey]cachedSubringWithLookback[*PartitionRing]
+}
+
+func newPartitionRingShuffleShardCache() *partitionRingShuffleShardCache {
+	return &partitionRingShuffleShardCache{
+		cacheWithoutLookback: map[subringCacheKey]*PartitionRing{},
+		cacheWithLookback:    map[subringCacheKey]cachedSubringWithLookback[*PartitionRing]{},
+	}
+}
+
+func (r *partitionRingShuffleShardCache) setSubring(identifier string, size int, subring *PartitionRing) {
+	if subring == nil {
+		return
+	}
+
+	r.mtx.Lock()
+	defer r.mtx.Unlock()
+
+	r.cacheWithoutLookback[subringCacheKey{identifier: identifier, shardSize: size}] = subring
+}
+
+func (r *partitionRingShuffleShardCache) getSubring(identifier string, size int) *PartitionRing {
+	r.mtx.RLock()
+	defer r.mtx.RUnlock()
+
+	cached := r.cacheWithoutLookback[subringCacheKey{identifier: identifier, shardSize: size}]
+	if cached == nil {
+		return nil
+	}
+
+	return cached
+}
+
+func (r *partitionRingShuffleShardCache) setSubringWithLookback(identifier string, size int, lookbackPeriod time.Duration, now time.Time, subring *PartitionRing) {
+	if subring == nil {
+		return
+	}
+
+	var (
+		lookbackWindowStart                   = now.Add(-lookbackPeriod).Unix()
+		validForLookbackWindowsStartingBefore = int64(math.MaxInt64)
+	)
```

Comment:
nit: Don't you find this nicer and easier to read?

```suggestion
	lookbackWindowStart := now.Add(-lookbackPeriod).Unix()
	validForLookbackWindowsStartingBefore := int64(math.MaxInt64)
```

Reply:
I honestly do find it easier to read when `=` are aligned :)